src/lightning/pytorch/core/module.py

```diff
@@ -153,8 +153,8 @@ def optimizers(self, use_pl_optimizer: bool = True) -> MODULE_OPTIMIZERS:
 
         Args:
             use_pl_optimizer: If ``True``, will wrap the optimizer(s) in a
-                :class:`~lightning.pytorch.core.optimizer.LightningOptimizer` for automatic handling of precision and
-                profiling.
+                :class:`~lightning.pytorch.core.optimizer.LightningOptimizer` for automatic handling of precision,
+                profiling, and counting of step calls for proper logging and checkpointing.
```

Comment:
You could also add that the `.step()` method is specifically wrapped. So optimizers with a different name will not work: https://github.com/Lightning-AI/lightning/issues/16143